gateway.go

```diff
@@ -377,3 +404,23 @@ func stripClosingDot(s string) string {
 	}
 	return s
 }
+
+// src: https://github.com/coredns/coredns/blob/0aee758833cabb1ec89756a698c52b83bbbdc587/plugin/etcd/msg/service.go#L145
+// Split255 splits a string into 255 byte chunks.
+func split255(s string) []string {
```

Comment:
These functions, should it be placed in a utils.go?

Reply:
That could be the same for `isdns1123Hostname`

Reply:
Yup, to have less changes we could do the cleanup in a separate pr. All good

Reply:
Ok, I reseted to the previous one.